web/packages/teleterm/src/ui/HeadlessAuthn/HeadlessPrompt/HeadlessPrompt.tsx

```diff
@@ -0,0 +1,124 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import * as Alerts from 'design/Alert';
+import { ButtonIcon, Box, Text, ButtonSecondary } from 'design';
+import Dialog from 'design/Dialog';
+import { DialogContent, DialogHeader, DialogFooter } from 'design/Dialog';
+import { PromptWebauthn } from '../../ClusterConnect/ClusterLogin/FormLogin/PromptWebauthn';
+import { Attempt } from 'shared/hooks/useAsync';
+import * as Icons from 'design/Icon';
+
+export type HeadlessPromptProps = {
+  clientIp: string;
+  onApprove(): Promise<void>;
+  onReject(): Promise<void>;
+  updateHeadlessStateAttempt: Attempt<void>;
+  onCancel(): void;
+  onSuccess(): void;
+};
+
+export function HeadlessPrompt({
+  clientIp: clientIp,
+  onApprove,
+  onReject,
+  updateHeadlessStateAttempt,
+  onCancel,
+  onSuccess,
+}: HeadlessPromptProps) {
+  const [waitForMfa, setWaitForMfa] = useState(false);
+
+  return (
+    <Dialog
+      dialogCss={() => ({
+        maxWidth: '480px',
+        width: '100%',
+        padding: '0',
+      })}
+      disableEscapeKeyDown={false}
+      open={true}
+    >
+      <Box p={4}>
+        <DialogHeader
+          justifyContent="space-between"
+          mb={0}
+          alignItems="baseline"
+        >
+          <Text typography="h4" bold>
+            Headless Login Detected
+          </Text>
+          <ButtonIcon
+            type="button"
+            onClick={onCancel}
+            color="text.slightlyMuted"
+          >
+            <Icons.Close fontSize={5} />
+          </ButtonIcon>
+        </DialogHeader>
+        {updateHeadlessStateAttempt.status === 'error' && (
+          <Alerts.Danger mb={0}>
+            {updateHeadlessStateAttempt.statusText}
+          </Alerts.Danger>
+        )}
+
+        {!waitForMfa && (
+          <>
+            <DialogContent>
+              <Text typography="body1" color="text.slightlyMuted">
+                Someone has initiated a command from {clientIp}. If it was not
+                you, click Reject and contact your administrator.
+              </Text>
+            </DialogContent>
+            <DialogFooter>
+              <ButtonSecondary
+                autoFocus
+                mr={3}
+                type="submit"
+                onClick={async e => {
+                  e.preventDefault();
+                  setWaitForMfa(true);
+                  await onApprove();
+                  onSuccess();
```

Comment:
We shouldn't call `onSuccess` unconditionally. 
I had some cluster configuration issues so the MFA requests were always failing, but I didn't see any errors in the dialog, it just closed.

IMO we should call `onSuccess` in `handleHeadlessApprove` and `handleHeadlessReject` in `HeadlessAuthentication`. Currently, we pass the function to `HeadlessPrompt` component, but actually we don't need it there - the logic can be kept in the parent. 
After making this change, we can check for errors:
```tsx
  async function handleHeadlessApprove(): Promise<void> {
    const [, error] = await updateHeadlessState(
      HeadlessAuthenticationState.HEADLESS_AUTHENTICATION_STATE_APPROVED
    );
    if (!error) {
      props.onSuccess();
    }
  }
```